daemon/delete.go

```diff
@@ -128,7 +128,6 @@ func (daemon *Daemon) cleanupContainer(container *container.Container, forceRemo
 		return errors.Wrapf(err, "unable to remove filesystem for %s", container.ID)
 	}
 
-	daemon.nameIndex.Delete(container.ID)
```

Comment:
Maybe I missed it, but I didn't see this being replaced by something else.

Missing a `if err := daemon.containerReplica.ReleaseName(...); err != nil {...}` here?